internal/data/hosts/hosts.go

```diff
@@ -33,6 +35,11 @@ func (r *hostsRepo) Save(ctx context.Context, model *biz.Host) (*biz.Host, error
 		return nil, nil
 	}
 
+	// Set a default value for workspace if it's not set
+	if model.Metadata.Workspace == "" {
+		model.Metadata.Workspace = "default"
```

Comment:
It may just a placeholder, but a note: object ids in Zanzibar are global, so there would only be one "default" workspace shared by all orgs, which probably isn't intended.

Reply:
Right, it's not a placeholder, I was proposing to merge it as is. The other option is to not set the relationship if the workspace is not passed.

What do you recommend?

cc @alechenninger

Reply:
So for hosted we need this to pull from the user's auth method. Basically, the default workspace for a resource needs to be based on the user's org ID.

I imagine something like this:

- As I understand there is some authentication abstraction in Inventory, the output of this should have, among other things, a notion of the user's "tenant" or "organization." ("Tenant" is what is used in RBAC... I would reuse that for consistency since that will be moved under project-kessel as well)
- At some point then in processing the create (probably outside of the repository?), if a workspace is not provided the default should be derived from the user's tenant.
- The current logic we have in RBAC is that the default workspace ID is exactly equal to the user's org ID. This may change before production but for now this is a fine starting point I think (and both RBAC and Inventory will agree, which is an important requirement)

Does this makes sense?

Reply:
@alechenninger Currently, inventory does not capture the org id.  Question, what if the reporter, e.g. HBI, is multi-tenant. We can not pull the org ID from the authN of the reporter to get the org ID for the resource.